client/src/services/fridge/fridge.service.ts

```diff
@@ -1,43 +1,37 @@
-import { Injectable } from '@angular/core';
+import { Inject, Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { Observable } from 'rxjs';
 
 export interface Ingredient {
+  id: number;
   name: string;
 }
 
-//Will be replaced with API, using for testing
-export const INGREDIENTS: Ingredient[] = [
-  {name: 'Really long ingredient name' },
-  {name: 'Chicken' },
-  {name: 'Rice' },
-  {name: 'Apple' },
-  {name: 'Lettuce' },
-  {name: 'Beans' },
-  {name: 'Potato' },
-  {name: 'Salt' },
-  {name: 'Sugar' },
-  {name: 'Flour' },
-  {name: 'Eggs' },
-  {name: 'Pepper' },
-];
-
-
 @Injectable({
   providedIn: 'root',
 })
 export class FridgeService {
-  constructor() {}
 
-  public getAvailableIngredients(): string[] {
-    //TODO: Replace when Backend API available
-    return [...INGREDIENTS].map((ingredient) => ingredient.name).sort();
+  public apiIngredientsList: Ingredient[] = [];
+
+  constructor(private http: HttpClient, @Inject('BASE_URL') private baseUrl: string) {
   }
 
-  public getLocallyStoredIngredients(): Set<string> {
-    const localIngredients = localStorage.getItem('userIngredients');
-    if (localIngredients != null) {
+  getIngredientsData(): Observable<Ingredient[]> 
+  {
+    return this.http.get<Ingredient[]>(this.baseUrl + 'api/fridge');
```

Comment:
Once you replace httpClient with httpClientService, replace this line too to call the httpClientService's get method instead and only pass in for the url "fridge". httpClientService deals with the 'api/' route separately